lib/lemon_docs.rb

```diff
@@ -18,21 +20,26 @@ def self.parse(raw_blueprint)
     serialize_options[:sourcemap] = 1
     serialize_options[:drafter_format] = 1
 
-    if LemonDocs::Binding.drafter_check_blueprint(raw_blueprint, validation_result, 0) == 0
+    output = if LemonDocs::Binding.drafter_check_blueprint(raw_blueprint, validation_result, 0) == 0
       LemonDocs::Binding.drafter_parse_blueprint_to(raw_blueprint, parse_result, 0, serialize_options)
       json_output(parse_result)
     else
       json_output(validation_result)
     end
-
+    output
   ensure
     LemonDocs::Memory.free(parse_result)
     LemonDocs::Memory.free(validation_result)
   end
 
   def self.parse_file(file_path)
     file = File.new(file_path)
-    parse(file.read)
+    content = file.read
+    output = parse(content)
+    if LemonDocs::Validator.validate!(output)
+      markdown = Redcarpet::Markdown.new(Redcarpet::Render::HTML).render(content)
+      IO.write(file_path.gsub(".md", ".html"), markdown)
+    end
```

Comment:
[1] pry(main)> require_relative 'lib/lemon_docs'
=> true
[2] pry(main)> LemonDocs.parse_file("/home/berlin/md/test.md")

you will get /home/berlin/md/test.html